### DISCUSSION TITLE
Should deployment modules be seperate repositories?

### DISCUSSION THREAD

**Discussion by GeoWill** [developmentseed/titiler#562] [General]
First of all thanks for the amazing project, and the great speed of development. It can be hard to keep up! 
I've found the work put into deployment particularly helpful - particularly managing the production of AWS lambda functions with all the necessary spatial libraries. 

For context, I'm running a few different instances of TiTiler for different projects. They're all quite small, but by splitting them between AWS accounts it's easy to understand where the costs are. To do this I end up having several local checkouts of TiTiler. I was just thinking about moving these checkouts to be submodules of the various project modules, so that I can include some additional aws resources - route53 etc, and scripts. 
This made me think that having just the deployment repo as the submodule would be nice. It might also make sense in terms of organising the code, issues and discussions. On the other hand it might just be more admin and opportunities to find git confusing. 

If you thought it was a good idea I'd be happy to help update the docs with an example of how to add one of the deployment repos as a submodule of an existing project.

**Comment by vincentsarago** [comment]
Hi @GeoWill, 
thanks for starting the discussion. 

To be honest the `deployment` part is here for `documentation` purpose (we also use it to deploy `titiler.xyz` endpoint but because it's the public demo for titiler). 

>  To do this I end up having several local checkouts of TiTiler. 

do you mean you have multiple copy of the titiler code base? or you have multiple copies of the cdk code? 

> This made me think that having just the deployment repo as the submodule would be nice. 

I don't think we want to have this because then every org could come and ask for specific deployment options/cloud providers ....

**Comment by GeoWill** [comment]
I ended up having several copies of the titiler codebase. I think it probably makes sense for me to just grab the cdk code and integrate it into my own stacks. Makes total sense keeping a small number of examples in the main repo. :+1:

**Reply by vincentsarago** [reply]
> I ended up having several copies of the titiler codebase. 

Well 2 things:
- TiTiler is a proper python module so `forking` is not necessary 
- the CDK app doesn't use the titiler python code but install it from pypi https://github.com/developmentseed/titiler/blob/650f6e45e3ad097360abf2e639b643cce83d3e45/deployment/aws/lambda/Dockerfile#L8 

As you said, you only need to copy/duplicate some of https://github.com/developmentseed/titiler/tree/master/deployment/aws
